src/libraries/Hooks.sol

```diff
@@ -236,22 +232,22 @@ library Hooks {
     /// @notice calls beforeSwap hook if permissioned and validates return value
     function beforeSwap(IHooks self, PoolKey memory key, IPoolManager.SwapParams memory params, bytes calldata hookData)
         internal
-        returns (int256 amountToSwap, BeforeSwapDelta hookReturn)
+        returns (int256 amountToSwap, BeforeSwapDelta hookReturn, uint24 lpFee)
     {
         amountToSwap = params.amountSpecified;
-        if (msg.sender == address(self)) return (amountToSwap, BeforeSwapDeltaLibrary.ZERO_DELTA);
+        lpFee = LPFeeLibrary.NO_FEE_UPDATE_FLAG;
```

Comment:
maybe no fee override is a more appropriate name? cause it doenst actually update the cache